pyomo/contrib/community_detection/detection.py

```diff
@@ -0,0 +1,156 @@
+"""Community Detection Code - Rahul Joglekar"""
+from pyomo.core import ConcreteModel
+from contrib.community_detection import community_graph
+import community
```

Comment:
This is a point that we haven't discussed yet but that seems important. What happens if the user does not have the package `community` installed? We should have a way of checking this import and besides hopefully having this package installed along with Pyomo.
Follow the code here https://github.com/Pyomo/pyomo/blob/07d3cdbde1139d3f926d322132c1995e00187b62/pyomo/contrib/mindtpy/mip_solve.py#L28 to attempt an import of the package and add the documentation of the package in the `communities.rst` file